packages/platform-ios/src/commands/runIOS/index.js

```diff
@@ -66,65 +71,37 @@ function runIOS(_: Array<string>, ctx: ConfigT, args: FlagsT) {
     }),
   );
 
-  if (args.device) {
-    const selectedDevice = matchingDevice(devices, args.device);
+  if (args.device || args.udid) {
+    const selectedDevice = args.device
+      ? matchingDevice(devices, args.device)
+      : matchingDeviceByUdid(devices, args.udid);
+
     if (selectedDevice) {
-      return runOnDevice(
-        selectedDevice,
-        scheme,
-        xcodeProject,
-        args.configuration,
-        args.packager,
-        args.verbose,
-        args.port,
-      );
+      return runOnDevice(selectedDevice, scheme, xcodeProject, args);
     }
+
     if (devices && devices.length > 0) {
-      // $FlowIssue: args.device is defined in this context
-      logger.error(`Could not find device with the name: "${args.device}".
-Choose one of the following:${printFoundDevices(devices)}`);
-    } else {
-      logger.error('No iOS devices connected.');
+      const device = ((args.device: any): string);
+      const udid = ((args.udid: any): string);
```

Comment:
Can you move device and uuid up in the scope so that line 74 can also use them? Flow should understand it better and you likely won't need to cast like this

Reply:
Was going to ask about this, didn't know what it was the best to do and I followed what's in https://flow.org/en/docs/types/casting/#toc-type-casting-through-any.

What do you mean by moving it up in the scope? Do you mean moving the lines 84 and 85 to before 74?

Reply:
yes

Reply:
But as far as line 74 is concerned the `args.device` and `args.udid` can still be other types rather than `string`, why is it better to put it before it?

Reply:
If the object property is bound to variable Flow knows that it's not gonna mutate in flight (it's a bug in Flow imo, but whatever). So by doing that Flow will understand your if branch better and will know that `uuid` and `device` cannot be undefined in your ternary. At least that's what I would expect

Reply:
Alright, done!